lib/mongoose_push_web/plugs/maybe_render_spec.ex

```diff
@@ -0,0 +1,28 @@
+defmodule MongoosePushWeb.Plug.MaybeRenderSpec do
+  import Plug.Conn
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts) do
+    enable = Application.get_env(:mongoose_push, :openapi)[:expose_spec]
+
+    opts =
+      if enable do
+        OpenApiSpex.Plug.RenderSpec.init(opts)
+      end
+
+    %{enabled: enable, opts: opts}
+  end
+
+  @impl Plug
+  def call(conn, %{enabled: enable, opts: opts}) when enable == false or opts == nil do
+    conn
+    |> send_resp(404, "swagger.json access disabled")
```

Comment:
This plug returns JSON by default. If we want to return a text, we should make sure the content type and format of the response is correct.